src/color_gradients.jl

```diff
@@ -1,9 +1,74 @@
+type ColorLibrary
+    defaults::Dict{Symbol, Symbol}
+    lib::Dict{Symbol, Vector{RGBA{Float64}}}
+    ColorLibrary(defaults = Dict(:default => :sequential), lib = Dict{Symbol, Vector{RGBA{Float64}}}()) = new(defaults, lib)
+end
+
+ColorLibrary(lib::Dict{Symbol, Vector{RGBA{Float64}}}) =
+    ColorLibrary(Dict(:default => keys(lib)[1]), lib)
+
+function ColorLibrary(lib::Dict{Symbol, Vector{RGBA{Float64}}}, default::Symbol)
+    in(default, keys(lib)) || error("There is no gradient named $default in lib")
+    ColorLibrary(Dict(:default => default), lib)
+end
+
+function cgraddefaults(cl::ColorLibrary; default = nothing, sequential = nothing, diverging = nothing)
+    default == nothing || (cl.defaults[:default] = default)
+    sequential == nothing || (cl.defaults[:sequential] = sequential)
+    diverging == nothing || (cl.defaults[:diverging] = diverging)
+end
+
+const color_libraries = Dict{Symbol, ColorLibrary}()
+
+function getgradient(gradient::Symbol = :default, clibrary::Symbol = _gradients[1])
+    haskey(color_libraries, clibrary) || error("There is no color library named $clibrary . Use clibraries() to get a list of available color libraries")
+    cl = color_libraries[clibrary]
+    getgradient(gradient, cl)
+end
+
+function getgradient(gradient::Symbol, cl::ColorLibrary)
+    while haskey(cl.defaults, gradient)
+        gradient = cl.defaults[gradient]
+    end
+    haskey(cl.lib, gradient) && return cl.lib[gradient]
+
+    potentials = [name for (name, library) in color_libraries if haskey(library.lib, gradient)]
+    length(potentials) == 0 && error("There is no gradient named $gradient . Use cgradients() to get a list of gradients in the current color library, clibraries() to get a list of available color libraries")
+    length(potentials) > 1 && warn("$gradient is found in more than one library: $(join(potentials, ", ")). Choosing $(potentials[1])")
+    color_libraries[potentials[1]][gradient]
+end
+
+getindex(cl::ColorLibrary, key::Symbol) = getgradient(key, cl)
+
```

Comment:
Might be nice with a `setindex` function as well to easily add gradients (quite useful for PlotThemes, I'd imagine)